src/main/java/org/jvnet/hudson/test/WarExploder.java

```diff
@@ -74,18 +78,31 @@ private static File explode() throws Exception {
             if(new File(d,".jenkins").exists()) {
                 File dir = new File(d,"war/target/jenkins");
                 if(dir.exists()) {
-                    System.out.println("Using jenkins.war resources from "+dir);
+                    LOGGER.log(Level.INFO, "Using jenkins.war resources from {0}", dir);
                     return dir;
                 }
             }
         }
 
         // locate jenkins.war
+        File war;
         URL winstone = WarExploder.class.getResource("/winstone.jar");
-        if(winstone==null)
-            // impossible, since the test harness pulls in jenkins.war
-            throw new AssertionError("jenkins.war is not in the classpath. If you are running this from the core workspace, run 'mvn install' to create the war image in war/target/jenkins. If running from a plugin, try using `mvn clean test` (cf. JENKINS-45245).");
-        File war = Which.jarFile(Class.forName("executable.Executable"));
+        if (winstone != null) {
+            war = Which.jarFile(Class.forName("executable.Executable"));
+        } else {
+            // JENKINS-45245: work around incorrect test classpath in IDEA. Note that this will not correctly handle timestamped snapshots; in that case use `mvn test`.
+            File core = Which.jarFile(Jenkins.class); // will fail with IAE if have neither jenkins-war.war nor jenkins-core.jar in ${java.class.path}
```

Comment:
s/IAE/IDEA/

Reply:
Nope, `IllegalArgumentException`.

Reply:
Might be worth spelling that out, then, 'cos `IDEA` a line above `IAE` is, as demonstrated here, potentially confusing. =)

Reply:
Seconded.